ngraph/core/src/op/range.cpp

```diff
@@ -46,6 +46,11 @@ bool ngraph::op::v4::Range::visit_attributes(AttributeVisitor& visitor)
 
 void op::v4::Range::validate_and_infer_types()
 {
+    NODE_VALIDATION_CHECK(this,
+                          m_output_type.is_integral_number() || m_output_type.is_real(),
+                          "output tensor type should be a numeric type. Got: ",
+                          m_output_type);
+
     set_input_is_relevant_to_shape(0);
```

Comment:
We have a task 44524, please add new tests to be sure that your PR cover cases from this ticket. Thanks!

Reply:
Done